### DISCUSSION TITLE
Browser extensions don't auto update

### DISCUSSION THREAD

**Discussion by ShlomiD83** [dani-garcia/vaultwarden#3600] [Q&A]
Hi all,
When using the testing images I noticed that the browser extension doesn't auto update on changes.
I confirmed that WS notifications work (via developer mode) and the web vault updates instantly.

Please advise.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.28.1-44e9e1a5
* Web-vault version: v2023.5.0
* OS/Arch: linux/aarch64
* Running within Docker: true (Base: Alpine)
* Environment settings overridden: true
* Uses a reverse proxy: false
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: false
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.41.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, INVITATIONS_ALLOWED, SHOW_PASSWORD_HINT, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_FROM_NAME, SMTP_USERNAME, SMTP_PASSWORD

```json
{
  "_duo_akey": "***",
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********************",
  "domain_origin": "*****://*********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": false,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "warn",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": true,
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "**************************",
  "smtp_from_name": "VaultWarden",
  "smtp_host": "*****************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "******",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Comment by BlackDex** [comment]
I just checked using the same Alpine Base Image, and that seems to work perfectly.
I tested by having the following opened.

1. Firefox web-vault
2. Firefox Extension
3. Chromium web-vault
4. Chromium extension
5. Bitwarden Desktop
6. Android Mobile Client

They all received the changed message, and they all updated the vault item near instant.

According to the above Support String, it looks like you are not using a reverse proxy. Is that true?
I doubt it, since you are using web-sockets, and probably still use port 3012 for this i think.

While i think it shouldn't cause much issues, but I'm not sure, it looks like you did not configure or set the `ip_header` to be forwarded to Vaultwarden. That might cause some strange issues, but in theory shouldn't.

Are there any error message in the vaultwarden logs during a cipher update?
What happens if you fully remove the extension and install it again?

**Comment by ShlomiD83** [comment]
I have no idea how but now it works <?>
I've changed nothing.

**Reply by BlackDex** [reply]
Hehe. Maybe a session/refresh-token expired and was renewed, which triggered a new connection?

**Comment by ShlomiD83** [comment]
The only thing I did is try different images and messing with the WS settings.

**Comment by ShlomiD83** [comment]
`While i think it shouldn't cause much issues, but I'm not sure, it looks like you did not configure or set the ip_header to be forwarded to Vaultwarden. That might cause some strange issues, but in theory shouldn't.`

"ip_header": "X-Real-IP", that's not the correct setting? It's the default one.

**Reply by BlackDex** [reply]
Well, according to Vaultwarden, there isn't any IP-Header send by the reverse proxy, that is what the issue is.
It tries to detect several common headers, but none were found.

**Reply by ShlomiD83** [reply]
what setting do you have setup on your config?
